src/store/flow/block-types/BaseBlockStore.ts

```diff
@@ -46,11 +47,9 @@ export const actions = {
           },
         },
       },
-    }, {
-      // Extra vendor_metadata from consumer side (eg: some configs under a new namespace)
-      vendor_metadata: await dispatch('initiateExtraVendorConfig'),
-    },
-    )
+    }
+
+    const mainProps = defaultsDeep({}, vendorMetadata, props, defaultProps)
```

Comment:
`_.defaults` method [https://devdocs.io/lodash~4/index#defaults](won't override) `vendorMetadata`, so I changed the order of assignments.